src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java

```diff
@@ -212,7 +212,7 @@ public void saveCaseEvent(String authorisation, Claim claim, CaseEvent caseEvent
 
     @Override
     public void updateClaimState(String authorisation, Long claimId, ClaimState state) {
-        if (state == ClaimState.OPEN) {
+        if (state == ClaimState.CREATE) {
```

Comment:
Also you don't need this check here as you already have this here (before calling this):
https://github.com/hmcts/cmc-claim-store/pull/995/files#diff-c18491d71318a89d1f0de872c9af15cdR117

Reply:
The main point of having the check here is to make sure nobody calls this method for state transition other than create -> open. So this check should stay here.

Reply:
agree with @karthikkannankainos on this.

Reply:
The state machine is controlled by CCD based on the event we fire, So the whole purpose of this method is to fire an event and CCD will change it's state based on the postcondition defined. CCD will check the precondition before applying this ISSUE_CLAIM event.

Reply:
This means we make another call to CCD which we can otherwise avoid.

Reply:
@karthikkannankainos I was told that state here passed here should be OPEN, then we call CLAIM_ISSUE Event, otherwise error, why we are changing to `CREATE`

Reply:
@dharmendrak the logic has changed here. Please see if the code change applies intact

Reply:
As per discussion with the reviewer, I have removed the exception being thrown. Also changed the all invoking code to send the current state of the claim this may cause issues if attempted to run in a non async mode.